extensions/java-joor-dsl/deployment/src/main/java/org/apache/camel/quarkus/dsl/java/joor/deployment/JavaJoorDslProcessor.java

```diff
@@ -96,6 +100,62 @@ void compileClassesAOT(BuildProducer<JavaJoorGeneratedClassBuildItem> generatedC
             generatedClass
                     .produce(new JavaJoorGeneratedClassBuildItem(className, nameToResource.get(className).getLocation(),
                             result.getByteCode(className)));
+            registerForReflection(reflectiveClass, lambdaCapturingTypeProducer,
+                    result.getClass(className).getAnnotation(RegisterForReflection.class));
+        }
+    }
+
+    private void registerForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<LambdaCapturingTypeBuildItem> lambdaCapturingTypeProducer,
+            RegisterForReflection annotation) {
+        if (annotation == null) {
+            return;
+        }
+
+        for (String lambdaCapturingType : annotation.lambdaCapturingTypes()) {
+            lambdaCapturingTypeProducer.produce(new LambdaCapturingTypeBuildItem(lambdaCapturingType));
+        }
+        boolean methods = annotation.methods();
+        boolean fields = annotation.fields();
+        boolean ignoreNested = annotation.ignoreNested();
+        boolean serialization = annotation.serialization();
+        boolean unsafeAllocated = annotation.unsafeAllocated();
+
+        if (annotation.registerFullHierarchy()) {
+            LOG.warn(
+                    "The element 'registerFullHierarchy' of the annotation 'RegisterForReflection' is not supported by the extension Camel Java DSL");
```

Comment:
"not supported by Camel Java DSL" ?

Isn't it maybe "not supported by Camel Java Joor DSL" ?